url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/CameraAgent.java

```diff
@@ -0,0 +1,554 @@
+		}
+		// }
+		mk = null;
+		timer.cancel();
+		timer = new Timer();
+		timer.scheduleAtFixedRate(new TimerTask() {
+			// Fires every 3.001 seconds.
+			@Override
+			public void run() {
+				print("Waking up");
+				stateChanged();
+			}
+		}, System.currentTimeMillis(), 3000);
+		// stateChanged();
+	}
+
+	@Override
+	public void msgTakePictureNestDone(NestGraphics nest, boolean done, NestGraphics nest2, boolean done2) {
+		print("Received msgTakePictureNestDone from graphics");
+		boolean found1 = false;
+		boolean found2 = false;
+		// synchronized (nests) {
+		for (MyNest n : nests) {
+			if (n.nest.nestGraphics == nest) {
+				// In v0 all parts are good parts
+				print("nest " + nests.indexOf(n) + " photographed");
+				n.numFilledSnapshot = n.nest.currentParts.size();
+
+				if (n.state == NestStatus.PHOTOGRAPHING) {
+					n.state = NestStatus.PHOTOGRAPHED;
+				} else if (n.state == NestStatus.RE_PHOTOGRAPHING_ONCE) {
+					n.state = NestStatus.RE_PHOTOGRAPHED_ONCE;
+				} else if (n.state == NestStatus.RE_PHOTOGRAPHING_TWICE) {
+					n.state = NestStatus.RE_PHOTOGRAPHED_TWICE;
+				}
+
+				if (found2) {
+					break;
+				}
+				found1 = true;
+			}
+			if (n.nest.nestGraphics == nest2) {
+				// In v0 all parts are good parts
+				print("nest " + nests.indexOf(n) + " photographed");
+				n.numFilledSnapshot = n.nest.currentParts.size();
+				if (n.state == NestStatus.PHOTOGRAPHING) {
+					n.state = NestStatus.PHOTOGRAPHED;
+				} else if (n.state == NestStatus.RE_PHOTOGRAPHING_ONCE) {
+					n.state = NestStatus.RE_PHOTOGRAPHED_ONCE;
+				} else if (n.state == NestStatus.RE_PHOTOGRAPHING_TWICE) {
+					n.state = NestStatus.RE_PHOTOGRAPHED_TWICE;
+				}
+
+				if (found1) {
+					break;
+				}
+				found2 = true;
+			}
+		}
+		// }
+		animation.release();
+		stateChanged();
+	}
+
+	@Override
+	public void msgTakePictureKitDone(KitGraphics k, boolean done) {
+		print("Received msgTakePictureKitDone from graphics");
+		mk.kitDone = done;
+		boolean passed = true;
+		ArrayList<Part> DummyParts = new ArrayList<Part>();
+
+		for (PartType type : mk.kit.partsExpected.getConfig().keySet()) {
+			int count = 0;
+			for (Part p : mk.kit.parts) {
+				if (p.type.equals(type) && !p.partGraphics.isInvisible()) {
+					count++;
+				}
+				if (p.partGraphics.isInvisible()) {
+					DummyParts.add(p);
+				}
+			}
+
+			for (int i = 0; i < DummyParts.size(); i++) {
+				print("Removing dummy part");
+				mk.kit.parts.remove(DummyParts.get(i));
+			}
+
+			if (count != mk.kit.partsExpected.getConfig().get(type)) {
+				passed = false;
+				break;
+			}
+		}
+		if (passed) {
+			print("Kit Inspection passed.");
+			mk.ks = KitStatus.DONE;
+
+		} else {
+			print("Kit Inspection failed.");
+			mk.ks = KitStatus.FAILED;
+		}
+		animation.release();
+		stateChanged();
+	}
+
+	/*
+	 * Scheduler (non-Javadoc)
+	 * 
+	 * @see agent.Agent#pickAndExecuteAnAction()
+	 */
+	@Override
+	public boolean pickAndExecuteAnAction() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 302-314 into a separate method.